src/base/bittorrent/session.h

```diff
@@ -330,6 +330,10 @@ namespace BitTorrent
         void setNetworkInterfaceName(const QString &name);
         QString networkInterfaceAddress() const;
         void setNetworkInterfaceAddress(const QString &address);
+#ifdef Q_OS_WIN
+        QString cachedGuid() const;
```

Comment:
Why do you need public setter/getter for it? Seems it's just implementation detail and shouldn't be accessed via Session interface.

Reply:
Am I on the right track now?

Reply:
My comment above was about public getter/setter only. You can still use `CachedSettingValue` member variable to read/write it.

Reply:
Is it mandatory to use CachedSettingValue or current method will do?

Reply:
Well, at least `CachedSettingValue` will save you from having to repeat the setting name. I don't insist, but @Chocobo1 most likely will.
Otherwise I would just use `SettingsStorage::loadValue()` and `SettingsStorage::storeValue()` directly without wrapping them into `cachedGuid()` and `setCachedGuid()`.